sdl_android/src/main/java/com/smartdevicelink/proxy/SdlProxyBase.java

```diff
@@ -3425,12 +3428,164 @@ public void run() {
+				throw new SdlException("There is no valid connection to SDL. sendSequentialRequests cannot be called until SDL has been connected.", SdlExceptionCause.SDL_UNAVAILABLE);
+			}
+		}
+
+		if (rpcs == null){
+			//Log error here
+			throw new SdlException("You must send some RPCs", SdlExceptionCause.INVALID_ARGUMENT);
+		}
+
+		int requestCount = rpcs.size();
+
+		// Break out of recursion, we have finished the requests
+		if (requestCount == 0) {
+			listener.onFinished();
+			return;
+		}
+
+		RPCRequest rpc = rpcs.get(0);
+		if (rpc.getCorrelationID() == null) {
+			rpc.setCorrelationID(CorrelationIdGenerator.generateId());
+		}
+
+		rpc.setOnRPCResponseListener(new OnRPCResponseListener() {
+			@Override
+			public void onResponse(int correlationId, RPCResponse response) {
+				if (response.getSuccess()) {
+					// success
+					rpcs.remove(0);
+					listener.onUpdate(rpcs.size());
+					try {
+						// recurse after successful response of RPC
+						sendSequentialRequests(rpcs, listener);
+					} catch (SdlException e) {
+						e.printStackTrace();
+					}
+				} else {
+					// fail
+					listener.onError(correlationId, response);
+				}
+			}
+		});
+
+		sendRPCRequestWithCustomListener(rpc);
+	}
+
+	/**
+	 * Takes a list of RPCRequests and sends it to SDL. Responses are captured through callback on OnMultipleRequestListener.
+	 * For sending requests synchronously, use sendSequentialRequests
+	 *
+	 * @param rpcs is the list of RPCRequests being sent
+	 * @param listener listener for updates and completions
+	 * @throws SdlException if an unrecoverable error is encountered
+	 */
+	@SuppressWarnings("unused")
+	public void sendRequests(List<RPCRequest> rpcs, final OnMultipleRequestListener listener) throws SdlException {
+
+		if (_proxyDisposed) {
+			throw new SdlException("This object has been disposed, it is no long capable of executing methods.", SdlExceptionCause.SDL_PROXY_DISPOSED);
+		}
+
+		SdlTrace.logProxyEvent("Application called sendRequests", SDL_LIB_TRACE_KEY);
+
+		synchronized(CONNECTION_REFERENCE_LOCK) {
+			if (!getIsConnected()) {
+				SdlTrace.logProxyEvent("Application attempted to call sendRequests without a connected transport.", SDL_LIB_TRACE_KEY);
+				throw new SdlException("There is no valid connection to SDL. sendRequests cannot be called until SDL has been connected.", SdlExceptionCause.SDL_UNAVAILABLE);
+			}
+		}
+
+		if (rpcs == null || rpcs.size() == 0){
+			//Log error here
+			throw new SdlException("You must send some RPCs", SdlExceptionCause.INVALID_ARGUMENT);
+		}
+
+		for (int i = 0; i < rpcs.size(); i++) {
+			RPCRequest rpc = rpcs.get(i);
+			if (rpc.getCorrelationID() == null) {
+				rpc.setCorrelationID(CorrelationIdGenerator.generateId());
+			}
+			listener.addCorrelationId(rpc.getCorrelationID());
+			rpc.setOnRPCResponseListener(listener.getSingleRpcResponseListener());

```

Comment:
I don't believe this is the right thing to do. If the developer has already added a listener to each of these RPCs this will overwrite them and will prevent a developer's logic from being executed.

Reply:
Isn't the goal of sending everything at once is that the responses come in on the overridden `onResponse` method in the required `OnMultipleRequestListener`? This allows that.